staking/Dockerfile

```diff
@@ -6,9 +6,11 @@
 
 FROM projectserum/build:v0.27.0@sha256:0e11aced57c448c7da9bf0a563a146398ac8b88f357b8fc2e1be314b42320686
 
+ARG TEST=false
+ENV TEST_ENV=$TEST
 WORKDIR /workspace
 
 COPY . .
 
 CMD ["bash", "-c", \
-        "anchor build && cp target/deploy/staking.so /artifacts/staking.so"]
+        "if [ \"$TEST_ENV\" = \"true\" ]; then anchor build -- --features mock-clock; else anchor build; fi && if [ \"$TEST_ENV\" = \"true\" ]; then chmod -R 777 target/; fi && cp -r target/ /artifacts"]
```

Comment:
is the `chmod` needed here?

Reply:
Yes we run into permission errors without it. because the files are created by the docker user.